### PR TITLE
[Compiler Add] add math methods

### PR DESCRIPTION
**Pull request by Jim8y**
This pr adds Math methods: Clamp,  DivRem, and BigMul. And also their edge-case covered tests.

### CODE REVIEW

File: src/Neo.Compiler.CSharp/MethodConvert/System/SystemCall.cs [hunk 1]
```diff
@@ -427,6 +427,322 @@ private bool TryProcessSystemMethods(SemanticModel model, IMethodSymbol symbol,
...
+                {
+                    if (instanceExpression is not null)
+                        ConvertExpression(model, instanceExpression);
+                    if (arguments is not null)
+                        PrepareArgumentsForMethod(model, symbol, arguments);
+                    JumpTarget endTarget = new();
+                    AddInstruction(OpCode.MUL);
+                    AddInstruction(OpCode.DUP);
+                    Push(long.MinValue);
+                    Push(new BigInteger(long.MaxValue) + 1);
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/MethodConvert/System/SystemCall.cs:440]:
```suggestion
                    Push(int.MinValue);
                    Push(new BigInteger(int.MaxValue) + 1);
```

File: tests/Neo.Compiler.CSharp.TestContracts/Contract_Math.cs [hunk 2]
```diff
@@ -23,5 +23,90 @@ public static int abs(int a)
         {
             return Math.Abs(a);
         }
+
+        public static long bigMul(int a, int b)
+        {
+            return Math.BigMul(a, b);
+        }
+
+        public static (byte Quotient, byte Remainder) DivRemByte(byte left, byte right)
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.TestContracts/Contract_Math.cs:32]: we should use lowercase for arguments (all of them)

File: tests/Neo.Compiler.CSharp.UnitTests/UnitTest_Math.cs [hunk 3]
```diff
@@ -45,5 +48,406 @@ public void abs_test()
...
+
+        [TestMethod]
+        [ExpectedException(typeof(TestException))]
+        public void TestDivRemUlong_DivideByZero()
+        {
+            Contract.DivRemUlong((ulong)10, (ulong)0);
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(TestException))]
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/UnitTest_Math.cs:208]: I think that the `AreEqual` are not executed

File: tests/Neo.Compiler.CSharp.UnitTests/UnitTest_Math.cs [hunk 4]
```diff
@@ -45,5 +48,406 @@ public void abs_test()
...
+            ulong alternatingBits = 0xAAAAAAAAAAAAAAAA; // 1010...1010
+            var result = Contract.DivRemUlong(alternatingBits, 3);
+            var expected = Math.DivRem(alternatingBits, 3);
+            Assert.AreEqual(expected.Remainder, checked((ulong)(BigInteger)result[0]));
+            Assert.AreEqual(expected.Quotient, checked((ulong)(BigInteger)result[1]));
+            Assert.AreEqual(1110330, Engine.FeeConsumed.Value);
+        }
+
+
+        [TestMethod]
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/UnitTest_Math.cs:333]:
```suggestion
        [TestMethod]
```

**Review by shargon — APPROVED** [review]
